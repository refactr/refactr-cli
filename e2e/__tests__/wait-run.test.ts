/* eslint-disable @typescript-eslint/no-non-null-assertion */

import isArray from 'lodash/isArray';
import { execute } from '../helpers/execute';
import knownIds from '../helpers/knowIds';

describe('refactrctl wait-run', () => {
  // Give platform time to bootstrap runner.
  jest.setTimeout(60 * 1000);

  let runId: string;
  beforeAll(async () => {
    const data = JSON.parse(
      await execute(
        [
          'run',
          'pipeline',
          '--project-id',
          knownIds.dynamic.project,
          '--revision-id',
          knownIds.dynamic.pipelineRevision,
          knownIds.dynamic.pipeline,
          '--format=json'
        ],
        { token: process.env.DYNAMIC_REFACTR_AUTH_TOKEN! }
      )
    );

    runId = data._id;
  });

  test('throws on missing arguments', async () => {
    await expect(
      execute(['wait-run'], {
        token: process.env.DYNAMIC_REFACTR_AUTH_TOKEN!
      })
    ).rejects.toMatchSnapshot();
  });

  test('should work', async () => {
    await expect(
      execute(
        [
          'wait-run',
          '--project-id',
          knownIds.dynamic.project,
          runId,
          '--format=json'
        ],
        { token: process.env.DYNAMIC_REFACTR_AUTH_TOKEN! }
      ).then((value) => isArray(JSON.parse(value)))
    ).resolves.toBeTruthy();
  });
});
