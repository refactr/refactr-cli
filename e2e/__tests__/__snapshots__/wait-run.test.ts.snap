// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refactrctl wait-run throws on missing arguments 1`] = `
"refactrctl wait-run [run-id]

Wait until run execution is finished

Positionals:
  run-id  ID of the run to wait  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Refactr API server  [string] [default: REFACTR_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: REFACTR_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project containing the run  [string] [required]

Missing required argument: project-id
"
`;
